utoipa-gen/tests/path_parameter_derive_actix.rs

```diff
@@ -164,3 +164,113 @@ fn derive_params_from_method_args_actix_success() {
+        "[0].required" = r#"true"#, "Parameter required"
+        "[0].deprecated" = r#"null"#, "Parameter deprecated"
+        "[0].schema.type" = r#""string""#, "Parameter schema type"
+        "[0].schema.format" = r#""date""#, "Parameter schema format"
+
+        "[1].in" = r#""path""#, "Parameter in"
+        "[1].name" = r#""end_date""#, "Parameter name"
+        "[1].description" = r#""End date filter""#, "Parameter description"
+        "[1].required" = r#"true"#, "Parameter required"
+        "[1].deprecated" = r#"null"#, "Parameter deprecated"
+        "[1].schema.type" = r#""string""#, "Parameter schema type"
+        "[1].schema.format" = r#""date-time""#, "Parameter schema format"
+    };
+}
+
+#[test]
+fn derive_path_with_date_params_explicit_ignored() {
+    mod mod_derive_path_with_date_params {
+        use actix_web::{get, web, HttpResponse, Responder};
+        use chrono::{DateTime, Utc};
+        use serde_json::json;
+        use time::Date;
+
+        #[utoipa::path(
+            responses(
+                (status = 200, description = "success response")
+            ),
+            params(
+                ("start_date", description = "Start date filter", format = Date),
+                ("end_date", description = "End date filter", format = DateTime),
+            )
+        )]
+        #[get("/visitors/v1/{start_date}/{end_date}")]
+        #[allow(unused)]
+        async fn get_foo_by_date(path: web::Path<(Date, String)>) -> impl Responder {
+            let (start_date, end_date) = path.into_inner();
+            HttpResponse::Ok()
+                .json(json!({ "params": &format!("{:?} {:?}", start_date, end_date) }))
+        }
+    }
+
+    #[derive(OpenApi, Default)]
+    #[openapi(paths(mod_derive_path_with_date_params::get_foo_by_date))]
+    struct ApiDoc;
+
+    let doc = serde_json::to_value(ApiDoc::openapi()).unwrap();
+    let parameters = doc
+        .pointer("/paths/~1visitors~1v1~1{start_date}~1{end_date}/get/parameters")
+        .unwrap();
+
+    common::assert_json_array_len(parameters, 2);
+    assert_value! {parameters=>
+        "[0].in" = r#""path""#, "Parameter in"
+        "[0].name" = r#""start_date""#, "Parameter name"
+        "[0].description" = r#""Start date filter""#, "Parameter description"
+        "[0].required" = r#"true"#, "Parameter required"
+        "[0].deprecated" = r#"null"#, "Parameter deprecated"
+        "[0].schema.type" = r#""string""#, "Parameter schema type"
+        "[0].schema.format" = r#""date""#, "Parameter schema format"
+
+        "[1].in" = r#""path""#, "Parameter in"
+        "[1].name" = r#""end_date""#, "Parameter name"
+        "[1].description" = r#""End date filter""#, "Parameter description"
+        "[1].required" = r#"true"#, "Parameter required"
+        "[1].deprecated" = r#"null"#, "Parameter deprecated"
+        "[1].schema.type" = r#""string""#, "Parameter schema type"
+        "[1].schema.format" = r#"null"#, "Parameter schema format"
```

Comment:
note this is probably the problem described in https://github.com/juhaku/utoipa/issues/693 - `format = DateTime` does not override the rust arg being of type `String`.

I am not sure if this is expected / desirable behaviour.

Reply:
Right, I believe this is at least one of the inconsistencies between the without `actix_web` or other framework and with.

I don't remember what was the original thought when implementing this but now I could consider this as undesired behavior. I reckon the order of recognizing parameters should first come from fn arguments if framework is supported and later they should be overridable with manual `params(...)` definition.